src/timer.c

```diff
@@ -177,11 +190,15 @@ static int __proc_timers(timer_link_t * thead)
  */
 static void timer_interrupt_callback(void)
```

Comment:
the comment on this function says
```
 *
 This function is called by the interrupt controller whenever 
 * compare == count.
```
I'd suggest either amending the comment, for example add
"also called when inserting into or removing from the timers list to improve handling timers with tiny delays" (if that's right?)
or moving the function body to a new "proc all timers and update next timer interrupt" function, and calling that instead of `timer_interrupt_callback`?

Reply:
I've amended the comment and also renamed the function to `timer_poll` to avoid confusion.